### PR TITLE
Make common functions cluster agnostic

### PR DESCRIPTION
**Pull request by Xenwar**
As we have them now, the common functions for upgrade workflow run against specific cluster. With the introduction of pivoting, the hard coded kubeconfig need to change.

This PR removes mentions of kubecofig files and parameters the secret, cluster and file names involved. 
It also introduce two functions for setting the kubeconfig towards the target cluster or the ephemeral cluster.

Individual test cases should export the KUBECONFIG variable to the correct cluster as needed.

### CODE REVIEW

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 1]
```diff
@@ -471,5 +463,14 @@ function verify_kubernetes_version_upgrade() {
...
     done
+}
+
+function point_to_target_cluster() {
+    kubectl get secrets "${TARGET_KUBECONFIG_SECRET}" -n "${NAMESPACE}" -o json | jq -r '.data.value' | base64 -d >"${TARGET_KUBECONFIG_FILE}"
+    export KUBECONFIG="${TARGET_KUBECONFIG_FILE}"
+
+}
 
+function point_to_ephemeral_cluster() {
```
Comment (Jaakko-Os) [inline scripts/feature_tests/upgrade/upgrade_common.sh:474]: My experience is that 'management' cluster is more commonly used

File: scripts/feature_tests/upgrade/upgrade_common.sh [hunk 2]
```diff
@@ -471,5 +463,14 @@ function verify_kubernetes_version_upgrade() {
             break
         fi
     done
+}
+
+function point_to_target_cluster() {
```
Comment (Jaakko-Os) [inline scripts/feature_tests/upgrade/upgrade_common.sh:468]: The test script calls this function, right?
The test script(s) change(s) should be included in this PR.

**Review by Jaakko-Os — CHANGES_REQUESTED** [review]

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 1 above)
Comment (Xenwar) [reply scripts/feature_tests/upgrade/upgrade_common.sh]: changed.

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 2 above)
Comment (Xenwar) [reply scripts/feature_tests/upgrade/upgrade_common.sh:468]: Indeed, the functions in here are called from other test scripts. 
The way you suggested, we have https://github.com/metal3-io/metal3-dev-env/pull/455
But, it got bigger and trying to create smaller PRs such this.

**Comment by maelk** [comment]
Do we expect any case where the original kubeconfig would not be in place or the target cluster would not have that name. i.e. is your goal to only run in CI or allow the tests to be run against any management and deployed cluster pair ? If only in CI, this lgtm

**Comment by Xenwar** [comment]
This is to be in CI

**Comment by Xenwar** [comment]
If you see any future use cases that can be handled before hand, we can make changes to this PR accordingly as well.

File: scripts/feature_tests/upgrade/upgrade_common.sh (hunk 2 above)
Comment (Jaakko-Os) [reply scripts/feature_tests/upgrade/upgrade_common.sh:468]: Small PRs help the review process significantly.
Then it's good to refer to #455 as there is a dependency.

**Review by maelk — APPROVED** [review]

**Pull request merged**